src/Model/ChangeStreamIterator.php

```diff
@@ -244,9 +246,19 @@ private function extractResumeToken($document)
             return $this->extractResumeToken($document->bsonSerialize());
         }
 
-        $resumeToken = is_array($document)
-            ? ($document['_id'] ?? null)
-            : ($document->_id ?? null);
+        if ($document instanceof Document) {
+            $resumeToken = $document->has('_id')
```

Comment:
I don't find the documentation on the behavior of `Document::get('_id')` when the key is not set? Reading your code I think it throw an error and doesn't return `null` like a a access to an undefined property of an object.

Reply:
`get` throws an exception if the key wasn't found. I'm not sure if we can make the assumption that an `_id` is always present, as users can pass their own pipeline to `watch` which may remove the `_id` field from results.

Reply:
Ok, correction based on Jeremy's comment below: the `_id` field will always be present, so I've removed the check for the field.

I'll note that `get` throws an exception instead of returning `null` to allow differentiating between a missing field and a field with an explicit `null` value.